ArticlesApp/ArticlesApp/Managers/DataManager.swift

```diff
@@ -4,33 +4,50 @@ enum DataManagerError: Error {
     case Unknown
     case FailedRequest
     case InvalidResponse
+    case SerializationError
 
 }
 
+enum Result<T> {
+    case Error(type: DataManagerError)
+    case Success(data: T)
+}
+
 final class DataManager {
-    func fetchArticles() {
+    func fetchArticles(completion: @escaping ((_ result: Result<[Article]>) -> Void)) {
         let url = API.BaseURL
+        let urlRequest = URLRequest(url: url)
+        let mapper = ArticleMapper()
+
+        let task = URLSession.shared.dataTask(with: urlRequest, completionHandler: { (data, response, error) in
+            if error != nil {
+                completion(Result.Error(type: .FailedRequest))
+                return
+            }
 
-        let task = URLSession.shared.dataTask(with: url) { (data, response, error) in
-            if let articles = try? JSONSerialization.jsonObject(with: data!, options: []) as! [[String : AnyObject]] {
-                    let json = """
-                    {
-                      "title": "Obama Offers Hopeful Vision While Noting Nation's Fears",
-                      "authors": "Graham Spencer",
-                      "date": "05/26/2014"
-                    }
-                    """.data(using: .utf8)!
-                    do {
-                        let myStruct = try JSONDecoder().decode(Article.self, from: json)
-                        print(myStruct)
-                    }
-                    catch {
-                        print(error)
-                    }
+            if let articlesList = mapper.map(data: data!) {
+                completion(Result.Success(data: articlesList))
             } else {
-                print("Could not deserialize JSON.")
+                completion(Result.Error(type: .SerializationError))
             }
-        }
+        })
+
         task.resume()
     }
 }
+
+protocol ArticleMappable {
+    func map(data: Data) -> [Article]?
+}
+
+struct ArticleMapper: ArticleMappable {
+
+    func map(data: Data) -> [Article]? {
+        do {
+            return try JSONDecoder().decode([Article].self, from: data)
+        }
+        catch  {
+            return nil
+        }
+    }
```

Comment:
When you're returning `nil` for a non successful try execution, you can use just `try?` instead.